### PR TITLE
[terra-button-group] Implemented border styles to buttons within the 'terra-button-group'.

### PR DESCRIPTION
**Pull request by AH106586Harika**
### Summary
Fixed- Implemented border styles for buttons within the 'terra-button-group'.

**What was changed:**
Implemented border styles for button focus.

**Why it was changed:**

### Testing

This change was tested using:

- [X] WDIO
- [ ] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [ ] Accessibility review
- [ ] Functional review

### Additional Details

**This PR resolves:**

[UXPLATFORM-10077](https://jira2.cerner.com/browse/UXPLATFORM-10077)

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

**Comment by supreethmr** [comment]
Screenshot 2023-12-19 at 11 24 14 AM
Blue highlight still shows up on focus , I think we need to have only dashed outline for button on Focus.

File: packages/terra-button/src/_variables.scss [hunk 1]
```diff
@@ -15,8 +15,7 @@ $terra-button-neutral-color-scheme: (
   box-shadow: var(--terra-button-box-shadow-neutral, none),
   color: var(--terra-button-color-neutral, #1c1f21),
   focus-background-color: var(--terra-button-focus-background-color-neutral, #dedfe0),
-  focus-border-color: var(--terra-button-focus-border-color-neutral, #c8cacb),
-  focus-box-shadow: var(--terra-button-focus-box-shadow-neutral, 0 0 1px 3px rgba(76, 178, 233, 0.5), 0 0 7px 4px rgba(76, 178, 233, 0.35)),
```
Comment (supreethmr) [inline packages/terra-button/src/_variables.scss:19]: Terra-button screenshots are updated incorrectly due to removal of these styles. : https://github.com/cerner/terra-core/pull/3998/files#diff-6629a708ca1f301df6d09a8ba8c94580d13450beab7f124cc12e923ee63048ee 
Focus styles are not visible on focused button !!

File: packages/terra-button-group/src/ButtonGroup.module.scss [hunk 2]
```diff
@@ -27,12 +27,17 @@
       opacity: var(--terra-button-group-button-disabled-opacity, 0.3);
     }
 
+    &:focus {
+      outline: var(--terra-button-group-button-focus-outline, 2px dashed #000);
```
Comment (supreethmr) [inline packages/terra-button-group/src/ButtonGroup.module.scss:31]: this change is more appropriate on button component not on button group. As button-group consumes button focus styles should be updated on button. we need make sure outline color are appropriate for different button variants.

Comment (AH106586Harika) [reply packages/terra-button-group/src/ButtonGroup.module.scss]: Done.

File: packages/terra-button/src/_variables.scss (hunk 1 above)
Comment (AH106586Harika) [reply packages/terra-button/src/_variables.scss:19]: Updated.

**Comment by saket2403** [comment]
IS the black border not required for fusion-theme?

**Comment by AH106586Harika** [comment]
> IS the black border not required for fusion-theme?

No, Supreeth suggested to make changes for default and lowlight themes.

**Review by saket2403 — APPROVED** [review]

File: packages/terra-button-group/src/clinical-lowlight-theme/ButtonGroup.module.scss [hunk 3]
```diff
@@ -13,8 +13,9 @@
 
     --terra-button-group-button-focus-background-color: #1a2023;
     --terra-button-group-button-focus-border-color: #181b1d;
-    --terra-button-group-button-focus-box-shadow: 0 0 1px 3px rgba(76, 178, 233, 0.5), 0 0 7px 4px rgba(76, 178, 233, 0.35);
-
+    --terra-button-group-button-focus-box-shadow: none;
+    --terra-button-group-button-focus-outline: 2px dashed #b2b5b6;
```
Comment (supreethmr) [inline packages/terra-button-group/src/clinical-lowlight-theme/ButtonGroup.module.scss:17]: this variable is no longer used in buttonGroup.module.scss

File: packages/terra-button-group/src/orion-fusion-theme/ButtonGroup.module.scss [hunk 4]
```diff
@@ -21,7 +22,7 @@
     --terra-button-group-button-selected-and-hover-color: #fff;
 
     --terra-button-group-button-selected-and-focus-background-color: #52585b;
-    --terra-button-group-button-selected-and-focus-border-color: #6f7477;
+    --terra-button-group-button-selected-and-focus-border-color: rgba(76, 178, 233, 0.3);
```
Comment (supreethmr) [inline packages/terra-button-group/src/orion-fusion-theme/ButtonGroup.module.scss:25]: why do we need change in focus color for fusion theme ?

**Review by supreethmr — CHANGES_REQUESTED** [review]
I think we need to add outline-offset as well to improve the visibility of  focus border Below screenshots are taken with outline-offset : 1px .

Screenshot 2024-01-02 at 6 33 54 PM

Screenshot 2024-01-02 at 6 36 06 PM

File: packages/terra-button-group/src/clinical-lowlight-theme/ButtonGroup.module.scss (hunk 3 above)
Comment (AH106586Harika) [reply packages/terra-button-group/src/clinical-lowlight-theme/ButtonGroup.module.scss:17]: As suggested below added outline and outline-offset  in mixins

File: packages/terra-button-group/src/orion-fusion-theme/ButtonGroup.module.scss (hunk 4 above)
Comment (AH106586Harika) [reply packages/terra-button-group/src/orion-fusion-theme/ButtonGroup.module.scss]: Reverted.

File: packages/terra-button/src/orion-fusion-theme/Button.module.scss [hunk 5]
```diff
@@ -30,6 +30,7 @@
     --terra-button-active-background-color-neutral: #9b9fa1;
     --terra-button-active-border-color-neutral: #6f7477;
     --terra-button-active-color-neutral: #000;
+    --terra-button-group-button-focus-outline: none;
```
[the remark this answers is not in the document]
Comment (AH106586Harika) [reply packages/terra-button/src/orion-fusion-theme/Button.module.scss]: Reverted.

File: packages/terra-button/src/Button.module.scss [hunk 6]
```diff
@@ -64,7 +64,7 @@
     }
 
     &:focus {
-      outline: none;
+      outline: var(--terra-button-focus-outline, 2px dashed #000);
```
[the remark this answers is not in the document]
Comment (AH106586Harika) [reply packages/terra-button/src/Button.module.scss:67]: Done. Please review.

File: packages/terra-button-group/src/ButtonGroup.module.scss [hunk 7]
```diff
@@ -51,8 +52,8 @@
     &.is-selected.is-focused {
       background-color: var(--terra-button-group-button-selected-and-focus-background-color, #52585b);
       border-color: var(--terra-button-group-button-selected-and-focus-border-color, #52585b);
-      box-shadow: var(--terra-button-group-button-selected-and-focus-box-shadow, 0 0 1px 3px rgba(76, 178, 233, 0.5), 0 0 7px 4px rgba(76, 178, 233, 0.35));
-      color: var(--terra-button-group-button-selected-and-focus-color, #fff);
+      box-shadow: var(--terra-button-group-button-selected-and-focus-box-shadow, none);
+      color: var(--terra-button-group-button-selected-and-focus-color, #fff);
```
Comment (supreethmr) [inline packages/terra-button-group/src/ButtonGroup.module.scss:56]:
```suggestion
      color: var(--terra-button-group-button-selected-and-focus-color, #fff);
```

File: packages/terra-button/CHANGELOG.md [hunk 8]
```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+* Removed
+   * Removed 'focus-box-shadow' property in terra-button.
```
Comment (supreethmr) [inline packages/terra-button/CHANGELOG.md:6]:
```suggestion
* Added
   * Added Dotted focus border for terra-button.
```

**Review by supreethmr — APPROVED** [review]

**Pull request merged**